### ISSUE TITLE
Master ignores root_dir config value on startup

### ISSUE THREAD

**Issue by max-arnold** [saltstack/salt#32578]
### Description of Issue/Question

Salt-master ignores `root_dir` and tries to access /var/cache directory.
### Setup

I run salt-master from virtualenv on local machine using my regular user account. Directory structure looks like this:

```
cloud-profiles/
etc/
pillar/
salt/
var/
cloud
master
minion
proxy
roster
```

I set `root_dir: ./` (`cachedir` is commented out) and run salt-master locally with the following command: 

```
salt-master -c ./
```

It used to work just fine on `2015.5.10` and fails on `2015.8.8.2` with the following exception:

```
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/master.py", line 769, in run
    self.key,
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/master.py", line 1459, in __init__
    self.local = salt.client.get_local_client(self.opts['conf_file'])
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/client/__init__.py", line 96, in get_local_client
    return LocalClient(mopts=opts, skip_perm_errors=skip_perm_errors)
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/client/__init__.py", line 136, in __init__
    self.key = self.__read_master_key()
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/client/__init__.py", line 162, in __read_master_key
    self.skip_perm_errors)
  File "/Users/user/.virtualenvs/salt/lib/python2.7/site-packages/salt/utils/verify.py", line 400, in check_path_traversal
    raise SaltClientError(msg)
SaltClientError: Could not access /var/cache/salt. Path does not exist.
```
### Versions Report

```
Salt Version:
           Salt: 2015.8.8.2

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
         Python: 2.7.11 (default, Dec  9 2015, 21:43:13)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist:
        machine: x86_64
        release: 15.4.0
         system: 10.11.4 -- x86_64
```

**Comment by jfindlay** [comment]
@max-arnold, thanks for the report.

**Comment by cachedout** [comment]
Does it work with `cachedir` set, then? I wouldn't expect it to otherwise, nor would I consider this a regression.

**Comment by max-arnold** [comment]
Setting `cachedir` to relative path `./var/cache/salt/master` doesn't work in `2015.8.8.2` and leads to the same exception. Setting `cachedir` to absolute path `/Users/user/salt/var/cache/salt/master` does work though.

Why this isn't a regression? This comment in master config file says it should respect `root_dir`:

```
# The root directory prepended to these options: pki_dir, cachedir,
# sock_dir, log_file, autosign_file, autoreject_file, extension_modules,
# key_logfile, pidfile.
```

**Comment by ritzk** [comment]
It does respect root_dir, but for some reason still uses cachedir during initial verification.

**Comment by max-arnold** [comment]
Still happens on 2016.11.6

**Comment by baby-gnu** [comment]
Hello.

In preparation for the solution to this issue we need to fix #51192.

Regards
